resources/js/components/Listing/ListingView.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+    <div>
+        <div v-if="initializing" class="card loading">
+            <loading-graphic />
+        </div>
+
+        <data-list
+            v-if="!initializing"
+            class="mb-4"
+            :visible-columns="columns"
+            :columns="columns"
+            :rows="items"
+            :sort="false"
+            :sort-column="sortColumn"
+            :sort-direction="sortDirection"
+        >
+            <div slot-scope="{ hasSelections }">
+                <div class="card p-0 relative">
+                    <data-list-filter-presets
+                        ref="presets"
+                        :active-preset="activePreset"
+                        :preferences-prefix="preferencesPrefix"
+                        @selected="selectPreset"
+                        @reset="filtersReset"
+                    />
+
+                    <div class="data-list-header">
+                        <data-list-filters
+                            :filters="filters"
+                            :active-preset="activePreset"
+                            :active-preset-payload="activePresetPayload"
+                            :active-filters="activeFilters"
+                            :active-filter-badges="activeFilterBadges"
+                            :active-count="activeFilterCount"
+                            :search-query="searchQuery"
+                            :saves-presets="true"
+                            :preferences-prefix="preferencesPrefix"
+                            @filter-changed="filterChanged"
+                            @search-changed="searchChanged"
+                            @saved="$refs.presets.setPreset($event)"
+                            @deleted="$refs.presets.refreshPresets()"
+                            @restore-preset="$refs.presets.viewPreset($event)"
+                            @reset="filtersReset"
+                        />
+                    </div>
+
+                    <div
+                        v-show="items.length === 0"
+                        class="p-3 text-center text-grey-50"
+                        v-text="__('No results')"
+                    />
+
+                    <data-list-table
```

Comment:
Is there any way we can make the first column a link to the 'Edit' page? Currently the only way for me seems to be opening the dropdown.

I'm guessing this is something to do with `has_link` in the `ResourceListingController`

Reply:
First column Edit is done.
Pagination - definitely there for me - try messing with line 38 in ResourceListingController?

Reply:
Thanks! Both things seem to work now.